### PR TITLE
G UI/map/4

### PR DESCRIPTION
**Pull request by osvegn**
Implement the visual map for the zappy, and update the tile class to fix some errors.

### CODE REVIEW

**Review by LucasTesnier — APPROVED** [review]
NTR!

**Review by GuyomT — APPROVED** [review]
NTR

File: src/zappy_gui_src/Core/SFML/Map/Map.cpp [hunk 1]
```diff
@@ -8,3 +8,73 @@
...
+}
+
+void Map::_updateTileVectorSize()
+{
+    while (_mapSize.x * _mapSize.y < _tile.size())
+        _tile.pop_back();
+    while (_mapSize.x * _mapSize.y > _tile.size()) {
+        _tile.push_back(std::make_unique<Tile>());
+        _tile.back()->setPosition(sf::Vector2f((_tile.size() - 1) % (int)_mapSize.x, (_tile.size() - 1) / (int)_mapSize.x));
+        _tile.back()->setSize(sf::Vector2f(100, 100));
```
Comment (pariciadra) [inline src/zappy_gui_src/Core/SFML/Map/Map.cpp:29]: What happen if setTexture will fail

**Review by pariciadra — COMMENTED** [review]
you should Check if set texture fail or not before display it on window or crash will occurre

File: src/zappy_gui_src/Core/SFML/Map/Map.cpp (hunk 1 above)
Comment (osvegn) [reply src/zappy_gui_src/Core/SFML/Map/Map.cpp:29]: if the image can't be load, the program won't crash. The color of the shape is set to white by default, and a message is print on the terminal.

Comment (osvegn) [reply src/zappy_gui_src/Core/SFML/Map/Map.cpp:29]: but @waroth have a compilation error, so I will check the return value

**Review by pariciadra — APPROVED** [review]
LGTM

**Pull request merged**